src/main/java/gregtech/common/covers/CoverFluidVoidingAdvanced.java

```diff
@@ -0,0 +1,192 @@
+package gregtech.common.covers;
+
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Cuboid6;
+import codechicken.lib.vec.Matrix4;
+import gregtech.api.cover.ICoverable;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.widgets.*;
+import gregtech.api.recipes.FluidKey;
+import gregtech.api.util.GTFluidUtils;
+import gregtech.api.util.GTHashMaps;
+import gregtech.client.renderer.texture.Textures;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.MathHelper;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+
+import java.util.HashMap;
+import java.util.function.Predicate;
+
+public class CoverFluidVoidingAdvanced extends CoverFluidVoiding {
+
+    protected VoidingMode voidingMode;
+    protected int transferAmount = 0;
+
+    public CoverFluidVoidingAdvanced(ICoverable coverHolder, EnumFacing attachedSide) {
+        super(coverHolder, attachedSide);
+        this.voidingMode = VoidingMode.VOID_ANY;
+    }
+
+    @Override
+    protected void doTransferFluids() {
+        IFluidHandler myFluidHandler = coverHolder.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, attachedSide);
+        if (myFluidHandler == null) {
+            return;
+        }
+        switch (voidingMode) {
+            case VOID_ANY:
+                GTFluidUtils.transferFluids(myFluidHandler, nullFluidTank, Integer.MAX_VALUE, fluidFilter::testFluidStack);
+            case VOID_OVERFLOW:
+                voidOverflow(myFluidHandler, fluidFilter::testFluidStack, this.transferAmount);
+        }
+    }
+
+    /**
+     * Performs one tick worth of Keep Exact behavior.
+     *
+     * @param sourceHandler source(s) to move fluids from
+     * @param fluidFilter   a predicate which determines what fluids may be moved
+     * @param keepAmount    the desired amount in milliBuckets of a particular fluid in the destination
+     */
+    protected void voidOverflow(final IFluidHandler sourceHandler,
```

Comment:
One thing that would be nice there would the the ability to set the `keepAmount` separately for different fluids. This can already be done for items, by changing the stack size of the item in the filter, but this cannot be done for fluids at all, so you end up applying just 1 number to all fluids, even though you may have wanted to limit the fluids to different amounts.

This was not really possible in CE, as it did not have the fluidstack stack size display in the slot, so you would never know what you were limiting the fluids to, but since we have that feature in CEu, the ability to set different fluid amounts in the fluid filter would be very appreciated. (If done for the fluid voiding cover, it should also be done for the Fluid Regulator, maybe at another time though)